### DISCUSSION TITLE
Customized Helm Values based on Destination Cluster Specific Data

### DISCUSSION THREAD

**Discussion by mvaalexp** [argoproj/argo-cd#12013] [General]
As the title suggests, I am looking for a way to customize Helm values based on cluster specific data.

Example use case:
I have a helm chart that needs to specify to AWS subnets as an annotation on a helm template resource.
I have many clusters and do not want the user to need to keep track of the subnets based on which cluster they are deploying to.

How do I allow users to template in values into their Helm chart (e.g., replace the value in their values.yam) based on known data in the destination cluster?

**Comment by jaideepr97** [comment]
Yeah i think this basically the `helm lookup` use case which is pretty common but unfortunately one that argo-cd doesn't support AFAIK 
I'm not aware of a feasible workaround at the moment though

**Reply by jaideepr97** [reply]
I will bring this up with the maintainers to see what they think

**Reply by mvaalexp** [reply]
Thank you. I am not sure this entirely covers it though.
One of the options our team was looking into was this plugin:
https://github.com/kuuji/helm-external-val

This would allow us to manipulate the configmap values prior to usage, but was hoping for a more standardized/supported approach.

`helm lookup` would also be useful (almost required) but this doesn't allow people to manipulate the input for pre-existing helm charts.
